### ISSUE TITLE
Hydra : Mammoth Hydra is consistently doing less dps in the sim 

### ISSUE THREAD

**Issue by plopinou** [d07RiV/d3planner#66]
The hydra does between 15 and 20% less in the sim than on the sheet, even with a perfect uptime and a spread of 0.
Even with only hydra as a selected/used spell, the difference remains.

Also, if you only have yellow gear without any crit or crit damage nor any gems/kanai's/skill except for hydra, the difference is "only" ~7% less dps.

I didn't find a specific factor that could explain this behavior, and as far as I know, hydra's in game are starting to DPS nearly instantaneously if casted on the mob's hitbox.

**Comment by d07RiV** [comment]
The sheet DPS is displayed as if all layers were constantly active, i.e. Hydra had an infinite duration.

In-game, however, when you cast Hydra, it starts stacking layers on the target,ramping up its damage. When the hydra expires, all its remaining layers are 'orphaned', and no longer benefit from your damage multipliers or CHC/CHD. This explains the difference between regular gear and a bunch of rares.

**Comment by plopinou** [comment]
But If you constantly recast the hydra before it despawns, like every 10s, the layers sould be constant, no ?
20% less damage is a lot for a 100% uptime hydra over a 5mn boss fight.

Does the ramping up takes around 3s and recasting the hydra instantly removes the damage benefit for the layer casted by the previous one ?

**Comment by dominiklohmann** [comment]
> But If you constantly recast the hydra before it despawns, like every 10s, the layers sould be constant, no ?

When you recast the hydra, it takes a while until it builds up all the layers again. And even then there might not be 100% uptime on "all" layers, but rather all except for one layer. This is why hydras deals the most damage to enemies less or equal than 5 yards away from them.

You also cannot recast it just in time, since replacing a hydra with another deletes the old hydras layers.

**Comment by d07RiV** [comment]
It doesn't instantly delete the old layers, but those remaining layers do a lot less damage. From what I understand, many effects have a static damage number, and dynamic modifiers that they inherit from the caster (to avoid snapshotting basically). But if the caster dies, they have no owner anymore and lose all these modifiers.

**Comment by plopinou** [comment]
I looked up a little more about the hydra mechanics, and now understand exactly how it's working.
To sum it up :
- once cast, the hydra will spawn a layer every FPA
- each layer will tick every 0.3s, and expand by 10 yards up to 50 yards (hence why it's better to cast at close range)
- after the 16th tick, the layer will despawn

So an hydra casted only once at 0 yard range will have an initial ramp-up of 4.8s, and the damage of the last layers will instantly go down by a lot once the hydra despawn if you have a high bonus multiplier like TR or FF.
So the actual average DPS of a single hydra should not be more than 84% of the sheet dps.

The question is, does recasting an hydra before the previous one expires still cancel the layer benefiting from buffs of the previous one ? This would mean that chain casting hydra would only let 1 to 3 ticks (depending on the FPA) being at full damage instead of 16, thus diving the dps by a lot.
I didn't find specific info on that, but I didn't know where to look for it. So I just tested it in game on Gom.

Result : I confirm that recasting an hydra just after another one cancel all the layer buffs from the previous one, whatever the distance from the mob. Chain casting hydra is a huge DPS loss.

So the sim is perfectly right, it was just my lack of knowledge. Thank you for your time !
